cpu/native/Makefile.dep

```diff
@@ -1,3 +1,6 @@
+ifneq (,$(filter periph_gpio,$(USEMODULE)))
+  USEMODULE += periph_gpio_linux
```

Comment:
always pulling `periph_gpio_linux` when `periph_gpio` is used will prevent almost all usage of native on non Linux systems because `periph_gpio` is always part of the build when provided by the platform:
https://github.com/RIOT-OS/RIOT/blob/a56a6a25ffcdfa9e5f85c06ca5b78d2113825053/Makefile.dep#L1065-L1066


I would only pull `periph_gpio_linux` if the OS is Linux.

There's the same problem with SPI but this peripheral is less problematic I think because it's not pulled by default.

Reply:
I've wrapped both inside a `ifeq ($(OS),Linux)` block